src/line_extractor_igvc/src/LineExtractorNode.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Created By: Min Gyo Kim
+ * Created On: January 20, 2018
+ * Description: Node implementation for Line Extractor Node
+ */
+
+#include <LineExtractorNode.h>
+
+LineExtractorNode::LineExtractorNode(int argc,
+                                     char** argv,
+                                     std::string node_name) {
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    std::string degree_polynomial_param = "degree_polynomial";
+    int default_degree_polynomial       = 3;
+    SB_getParam(private_nh,
+                degree_polynomial_param,
+                this->degreePoly,
+                default_degree_polynomial);
+
+    std::string lambda_param = "lambda";
+    float default_lambda     = 0;
+    SB_getParam(private_nh, lambda_param, this->lambda, default_lambda);
+
+    std::string min_neighbours_param = "min_neighbours";
+    int default_min_neighbours       = 3;
+    SB_getParam(private_nh,
+                min_neighbours_param,
+                this->minNeighbours,
+                default_min_neighbours);
+
+    std::string radius_param = "radius";
+    float default_radius     = 0.1;
+    SB_getParam(private_nh, radius_param, this->radius, default_radius);
+
+    if (areParamsInvalid()) { ros::shutdown(); }
```

Comment:
I really like this check! I would use a ROS_DEBUG though to let the user know what caused their node to shutdown.